src/ngraph/frontend/onnx_import/utils/reduction.hpp

```diff
@@ -0,0 +1,111 @@
+        {
+            namespace detail
+            {
+                inline AxisSet get_reduction_axes(const Node& node)
+                {
+                    auto reduction_axes =
+                        node.get_attribute_value<std::vector<std::size_t>>("axes", {});
+                    if (reduction_axes.empty())
+                    {
+                        reduction_axes = onnx_import::common::get_monotonic_range<std::size_t>(
+                            node.get_ng_inputs().at(0)->get_shape().size());
+                    }
+                    return AxisSet{reduction_axes};
+                }
+            } // namespace  detail
+
+            /// \brief      Create an nGraph version of an ONNX reduction operation.
+            ///
+            /// \param[in]  node          The node representing incoming ONNX operation.
+            ///
+            /// \tparam     OnnxOperator  Class of an nGraph ArithmeticReduction operation
+            ///                           (e.g. Min, Max, SUm, Product).
+            ///
+            /// \return     nGraph node equivalent of the ONNX operation.
+            ///
+            template <class OnnxOperator,
+                      typename std::enable_if<std::is_base_of<ngraph::op::util::ArithmeticReduction,
+                                                              OnnxOperator>::value,
+                                              int>::type = 0>
+            std::shared_ptr<ngraph::Node>
+                make_ng_reduction_op(const Node& node,
+                                     const std::shared_ptr<ngraph::Node>& ng_input)
+            {
+                auto data_shape = ng_input->get_shape();
+
+                auto reduction_axes = detail::get_reduction_axes(node);
+                if (reduction_axes.size() > data_shape.size())
+                {
+                    throw error::parameter::Value(node.op_type(),
+                                                  node.get_name(),
+                                                  "provided reduction axes count (" +
+                                                      std::to_string(reduction_axes.size()) +
+                                                      ") is larger than input tensor rank (" +
+                                                      std::to_string(data_shape.size()) + ")");
+                }
+                auto op_node = std::make_shared<OnnxOperator>(ng_input, reduction_axes);
+
+                std::int64_t keepdims = node.get_attribute_value<std::int64_t>("keepdims", 1);
+                if (keepdims == 0)
+                {
+                    return op_node;
+                }
+
+                auto output_shape = data_shape;
+                // flatten reduced axes
+                for (const auto& idx : reduction_axes)
+                {
+                    output_shape.at(idx) = 1;
+                }
+                auto reshaped_op_node = std::make_shared<ngraph::op::Reshape>(
```

Comment:
`return std::make_shared<ngraph::op::Reshape>(...);` the variable `reshape_op_node` is not required.

Reply:
The variable `reshape_op_node` makes the code more readable. Unless we're making an unnecessary copy here, let's keep it.

Reply:
The `return std::make_shared<ngraph::op::Reshape>()` clearly states that the function returns node which does reshape op.